jax/_src/scipy/optimize/line_search.py

```diff
@@ -298,11 +306,11 @@ def wolfe_two(dphi_i):
   def body(state):
     # no amax in this version, we just double as in scipy.
     # unlike original algorithm we do our next choice at the start of this loop
-    a_i = jnp.where(state.i == 1, 1., state.a_i1 * 2.)
+    a_i = jnp.where(state.i == 1, start_value, state.a_i1 * 2.)
     # if a_i <= 0 then something went wrong. In practice any really small step
     # length is a failure. Likely means the search pk is not good, perhaps we
     # are at a saddle point.
-    saddle_point = a_i < 1e-5
+    saddle_point = a_i < 1e-20
```

Comment:
1e-20 seems suspiciously small for float32

Reply:
Yep - higher thresholds were causing the optimizer to fail where it would have been able to converge given more time. However I was a bit wary of taking it out entirely (Chesterton's fence). I've deleted it in the updated commit, and can always bring it back if removing it does cause issues.